zcash_client_sqlite/src/scan.rs

```diff
@@ -188,6 +189,7 @@ pub fn scan_cached_blocks<P: AsRef<Path>, Q: AsRef<Path>>(
             let nf_refs: Vec<_> = nullifiers.iter().map(|(nf, acc)| (&nf[..], *acc)).collect();
             let mut witness_refs: Vec<_> = witnesses.iter_mut().map(|w| &mut w.witness).collect();
             scan_block(
+                &consensus::MainNetwork,
```

Comment:
This (and similar changes elsewhere) hard-codes `MainNetwork` into `zcash_client_sqlite`, which will break testnet usage. To correctly set this for the `zcash_client_sqlite` crate, add the following imports to `zcash_client_sqlite/src/lib.rs`:
```rust
#[cfg(feature = "mainnet")]
use zcash_primitives::MainNetwork as Network;

#[cfg(not(feature = "mainnet"))]
use zcash_primitives::TestNetwork as Network;
```

Then change this to:
```suggestion
            scan_block::<Network>(
```
and similarly everywhere else in `zcash_client_sqlite`.

Reply:
Hah... this is exactly what I was complaining about yesterday - using those compilation flags is dangerous!

Reply:
On the contrary; this was very easy to spot in review, because the type was explicit!